fluent-react/test/localized_render_test.js

```diff
@@ -69,6 +133,69 @@ foo
     ));
   });
 
+  test('overwrite existing attributes if allowed', function() {
+    const mcx = new MessageContext();
+    const l10n = new ReactLocalization([mcx]);
+
+    mcx.addMessages(`
+foo
+    .existing = ATTR
+`)
+
+    const wrapper = shallow(
+      <Localized id="foo" attrs={{existing: true}}>
+        <div existing={true} />
+      </Localized>,
+      { context: { l10n } }
+    );
+
+    assert.ok(wrapper.contains(
+      <div existing="ATTR" />
+    ));
+  });
+
+  test('protect existing attributes if setting is forbidden', function() {
+    const mcx = new MessageContext();
+    const l10n = new ReactLocalization([mcx]);
+
+    mcx.addMessages(`
+foo
+    .existing = ATTR
+`)
+
+    const wrapper = shallow(
+      <Localized id="foo" attrs={{existing: false}}>
+        <div existing={true} />
+      </Localized>,
+      { context: { l10n } }
+    );
+
+    assert.ok(wrapper.contains(
+      <div existing={true} />
+    ));
+  });
+
+  test('protect existing attributes by default', function() {
+    const mcx = new MessageContext();
+    const l10n = new ReactLocalization([mcx]);
+
+    mcx.addMessages(`
+foo
+    .existing = ATTR
+`)
+
+    const wrapper = shallow(
+      <Localized id="foo">
+        <div existing={true} />
+      </Localized>,
+      { context: { l10n } }
+    );
+
+    assert.ok(wrapper.contains(
+      <div existing={true} />
+    ));
+  });
+
```

Comment:
nice test coverage - makes the benefits of explicitly disallowing setting attrs clear.